lib/spack/spack/package.py

```diff
@@ -779,6 +779,15 @@ def cleanup():
                          "Manually remove this directory to fix:",
                          self.prefix)
 
+            if not (keep_prefix and keep_stage):
+                self.do_clean()
+            else:
+                tty.warn("Keeping stage in place despite error.",
+                         "Spack will refuse to uninstall dependencies of this package." +
+                         "Manually remove this directory to fix:",
+                         self.stage.path)
```

Comment:
If it's vestigial yes.  I think it's my bug if the thing was refusing to uninstall when a stage was present.  I'll take a look at this after you update the PR.